packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -74,52 +1543,37 @@ abstract class TreeDiagnosticsMixin {
     final StringBuffer result = new StringBuffer();
     result.write(toString());
     result.write(joiner);
-    final List<String> description = <String>[];
-    debugFillDescription(description);
-    result.write(description.join(joiner));
+    final List<DiagnosticsNode> properties = <DiagnosticsNode>[];
+    debugFillProperties(properties);
+    result.write(
+      properties
+        .where((DiagnosticsNode n) => !n.hidden)
+        .join(joiner));
     return result.toString();
   }
 
   /// Returns a string representation of this node and its descendants.
-  ///
-  /// This includes the information from [debugFillDescription], and then
-  /// recurses into the children using [debugDescribeChildren].
-  ///
-  /// The [toStringDeep] method takes arguments, but those are intended for
-  /// internal use when recursing to the descendants, and so can be ignored.
-  ///
-  /// See also:
-  ///
-  ///  * [toString], for a brief description of the object but not its children.
-  ///  * [toStringShallow], for a detailed description of the object but not its
-  ///    children.
+  @override
   String toStringDeep([String prefixLineOne = '', String prefixOtherLines = '']) {
-    String result = '$prefixLineOne$this\n';
-    final String childrenDescription = debugDescribeChildren(prefixOtherLines);
-    final String descriptionPrefix = childrenDescription != '' ? '$prefixOtherLines \u2502 ' : '$prefixOtherLines   ';
-    final List<String> description = <String>[];
-    debugFillDescription(description);
-    result += description
-      .expand((String description) => debugWordWrap(description, 65, wrapIndent: '  '))
-      .map<String>((String line) => "$descriptionPrefix$line\n")
-      .join();
-    if (childrenDescription == '') {
-      final String prefix = prefixOtherLines.trimRight();
-      if (prefix != '')
-        result += '$prefix\n';
-    } else {
-      result += childrenDescription;
-    }
-    return result;
+    return toDiagnosticsNode().toStringDeep(prefixLineOne, prefixOtherLines);
+  }
+
+  @override
+  DiagnosticsNode toDiagnosticsNode({String name, DiagnosticsTreeStyle style}) {
+    return new _TreeDiagnosticsMixinNode(
+        name: name, object: this, header: toString(), style: style);
   }
 
-  /// Add additional information to the given description for use by
-  /// [toStringDeep] and [toStringShallow].
+  /// Add additional properties for of the node.
```

Comment:
Agreed. Added a note that I will have to delete before the CL is complete (contains the text ZZZ) saying to write a many paragraph document on how to write such things. Don't want to write the doc until there is a bit more code review agreement on what the API looks like so I can write the doc once.